src/calendar/view/CalendarDayEventsView.ts

```diff
@@ -148,3 +149,10 @@ function getTimeIndicatorPosition(now: Date): number {
 	const passedMillisInDay = (now.getHours() * 60 + now.getMinutes()) * 60 * 1000
 	return (passedMillisInDay / DAY_IN_MILLIS) * allHoursHeight
 }
+
+function getTimeFromClickInteraction(e: MouseEvent, time: Time) {
+	const rect = (e.target as HTMLElement).getBoundingClientRect()
+	const mousePositionRelativeToRectHeight = Math.abs(rect.top - e.clientY)
+	if (mousePositionRelativeToRectHeight > rect.height / 2) return { ...time, minute: time.minute + 30 }
+	return time
+}
```

Comment:
```suggestion
function getTimeFromClickInteraction(e: MouseEvent, time: Time): Time  {
	const rect = (e.target as HTMLElement).getBoundingClientRect()
	const mousePositionRelativeToRectHeight = Math.abs(rect.top - e.clientY)
	if (mousePositionRelativeToRectHeight > rect.height / 2) return new Time(time.hour, time.minute + 30)
	return time
}
```
This function has quite a complicated return type; it can return a `Time` object or an object containing the hours and minutes. I would recommend simplifying by only returning one type.